File: leet_easy_c++/leet_167/leet_167a.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        unordered_map<int, int> umap;
        int tmp, dif, len = numbers.size();
        vector<int> sol(2,0);
        for(int i=0; i<len; i++){
            tmp = *(numbers.begin()+i);
            dif = target - tmp;
            if(umap.find(dif) != umap.end()){
                sol[0] = umap[dif] + 1;
                sol[1] = i + 1;
                return sol;
            }
            umap[tmp] = i;
        }
        return sol;
    }
};

File: leet_easy_python/leet_167/leet_167a.py
class Solution(object):
    def twoSum(self, numbers, target):
        p1 = 0
        p2 = len(numbers) - 1
        while p1 < p2:
            sum = numbers[p1] + numbers[p2]
            if sum == target: return [p1+1, p2+1]
            elif sum > target: p2 -= 1
            else: p1 += 1
            
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """